nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/recurrent/LSTMLayer.java

```diff
@@ -76,33 +74,39 @@
  */
 public class LSTMLayer extends DynamicCustomOp {
 
-    private LSTMConfiguration configuration;
+    private LSTMLayerConfig configuration;
 
     @Getter
-    private LSTMWeights weights;
+    private LSTMLayerWeights weights;
+
 
     public LSTMLayer() {
     }
 
-    public LSTMLayer(@NonNull SameDiff sameDiff, SDVariable maxTSLength, SDVariable x, SDVariable cLast, SDVariable yLast, LSTMWeights weights, LSTMConfiguration configuration) {
-        super(null, sameDiff, weights.argsWithInputs(maxTSLength, x, cLast, yLast));
+    public LSTMLayer(@NonNull SameDiff sameDiff, SDVariable maxTSLength, SDVariable x, SDVariable cLast, SDVariable yLast, LSTMLayerWeights weights, LSTMLayerConfig configuration) {
+        super(null, sameDiff, weights.argsWithInputs(x, maxTSLength, cLast, yLast));
         this.configuration = configuration;
         this.weights = weights;
-        addIArgument(configuration.iArgs(true));
-        addTArgument(configuration.tArgs());
+        addIArgument(iArgs());
+        addTArgument(tArgs(configuration));
+        addBArgument(bArgs(weights, maxTSLength, yLast, cLast, configuration));
+
+
     }
 
-    public LSTMLayer(INDArray x, INDArray cLast, INDArray yLast, INDArray maxTSLength, LSTMWeights lstmWeights, LSTMConfiguration lstmConfiguration) {
+    public LSTMLayer(INDArray x, INDArray cLast, INDArray yLast, INDArray maxTSLength, LSTMLayerWeights lstmWeights, LSTMLayerConfig LSTMLayerConfig) {
         super(null, null, lstmWeights.argsWithInputs(maxTSLength, x, cLast, yLast));
-        this.configuration = lstmConfiguration;
+        this.configuration = LSTMLayerConfig;
         this.weights = lstmWeights;
-        addIArgument(configuration.iArgs(true));
-        addTArgument(configuration.tArgs());
+        addIArgument(iArgs());
+        addTArgument(tArgs(configuration));
+
+        addBArgument(bArgs(weights, maxTSLength, yLast, cLast, configuration));
     }
 
     @Override
     public List<DataType> calculateOutputDataTypes(List<DataType> inputDataTypes) {
-        Preconditions.checkState(inputDataTypes != null && inputDataTypes.size() == 9, "Expected exactly 9 inputs to LSTMLayer, got %s", inputDataTypes);
+        Preconditions.checkState(inputDataTypes != null && inputDataTypes.size() == 8, "Expected exactly 8 inputs to LSTMLayer, got %s", inputDataTypes);
```

Comment:
Given the optional inputs for lstmLayer, we can have as few as 3 inputs (input, Wx, Wr only) or as many as 8 - or anywhere in between

https://github.com/eclipse/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/nn/recurrent/lstmLayer.cpp#L59-L96